cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -148,19 +143,18 @@ public void run() {
                                                                                   : Collections.emptySet();
 
       for (Goal goal : _detectionGoals) {
-        if (_loadMonitor.meetCompletenessRequirements(goal.clusterModelCompletenessRequirements())) {
+        if (_kafkaCruiseControl.loadMonitor().meetCompletenessRequirements(goal.clusterModelCompletenessRequirements())) {
           LOG.debug("Detecting if {} is violated.", goal.name());
           // Because the model generation could be slow, We only get new cluster model if needed.
           if (newModelNeeded) {
             if (clusterModelSemaphore != null) {
               clusterModelSemaphore.close();
             }
-            clusterModelSemaphore = _loadMonitor.acquireForModelGeneration(new OperationProgress());
+            clusterModelSemaphore = _kafkaCruiseControl.acquireForModelGeneration(new OperationProgress());
             // Make cluster model null before generating a new cluster model so the current one can be GCed.
             clusterModel = null;
-            clusterModel = _loadMonitor.clusterModel(now,
-                                                     goal.clusterModelCompletenessRequirements(),
-                                                     new OperationProgress());
+            clusterModel = _kafkaCruiseControl.clusterModel(goal.clusterModelCompletenessRequirements(),
```

Comment:
With this change, we do not need to pass `_time` to this class. Can we
1. drop _time from this class,
2. make the same change in other anomaly detectors?